### PR TITLE
Unit tests for the new agents monitoring logic in monitord

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#6458|

## Description

This PR creates the UT folder and project for **monitord**, and adds the corresponding unit tests for the new functions inside _monitord.c_ and _monitor_actions.c_

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [ ] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)
 
 
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/unit_tests/monitord/test_monitor_actions.c [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+char* __wrap_get_agent_id_from_name(__attribute__((unused)) char *agent_name) {
+    return mock_type(char*);
+}
+
+int __wrap_auth_remove_agent(__attribute__((unused)) int sock, const char *id, __attribute__((unused)) int json_format) {
+    check_expected(id);
+    return mock();
+}
+
+int __wrap_auth_connect() {
```
Comment (palaciosjeremias) [inline src/unit_tests/monitord/test_monitor_actions.c:44]: Shouldn't we create these wrappers inside an independent module?

File: src/unit_tests/monitord/test_monitor_actions.c [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+void test_monitor_send_disconnection_msg_agent_removed(void **state) {
+    char *agent = "Agent1-any";
+    char msg_to_send[OS_SIZE_1024];
+
+    expect_string(__wrap_wdb_find_agent, name, "Agent1");
+    expect_string(__wrap_wdb_find_agent, ip, "any");
+    will_return(__wrap_wdb_find_agent, -2);
+
+    // monitor_send_deletion_msg()
```
Comment (palaciosjeremias) [inline src/unit_tests/monitord/test_monitor_actions.c:163]: This commented code should be removed.

**Review by palaciosjeremias — APPROVED** [review]
LGTM

File: src/unit_tests/monitord/test_monitor_actions.c (hunk 1 above)
Comment (pereyra-m) [reply src/unit_tests/monitord/test_monitor_actions.c]: It was created!

File: src/unit_tests/monitord/test_monitor_actions.c (hunk 2 above)
Comment (pereyra-m) [reply src/unit_tests/monitord/test_monitor_actions.c]: Now the comment is **monitor_send_deletion_msg** without _()_ to make it clear that the test entered in another function

**Review by palaciosjeremias — APPROVED** [review]

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**